src/qt/qt_progsettings.cpp

```diff
@@ -43,24 +45,25 @@ ProgSettings::CustomTranslator* ProgSettings::translator = nullptr;
 QTranslator* ProgSettings::qtTranslator = nullptr;
 QString ProgSettings::getIconSetPath()
 {
-    QString roms_root;
-    if (rom_path[0])
-        roms_root = rom_path;
-    else {
-        roms_root = QString("%1/roms").arg(exe_path);
-    }
-
-    if (iconset_to_qt.isEmpty())
-    {
+    if (iconset_to_qt.isEmpty()) {
+        QVector<QFileInfo> roms_dirs;
+        // Walk rom_paths to get the candidates
+        for (rom_path_t *emu_rom_path = &rom_paths; emu_rom_path != nullptr; emu_rom_path = emu_rom_path->next) {
+            roms_dirs.append(QFileInfo(emu_rom_path->path));
+        }
```

Comment:
Is there any particular reason why you loop twice over the same elements?

Reply:
Now that you mention it, hmmm... 😄 

I guess my brain somehow separated the tasks into 1) get the dirs then 2) search each for icons. 

In case it makes a difference, this code block is only executed once per program run because `iconset_to_qt.isEmpty()` will return false after the first run.

It does seem to make more sense to combine the two.

Reply:
I combined the two into a single loop.